### ISSUE TITLE
Nested shorthand lists parsed as array

### ISSUE THREAD

**Issue by czosel** [glayzzle/php-parser#512]
Input:
```php
list($a, list($b, $c)) = array(1, array(2, 3));
[$a, [$b, $c]] = [1, [2, 3]];
```
The first line is parsed as `list` with a `list` inside, the second one as `list` with an `array` inside.

**Comment by czosel** [comment]
@ichiriac Can you share a pointer how to fix this? I might be able to take a look at it in the coming days.

**Comment by ichiriac** [comment]
@czosel released under 3.0.2

**Comment by czosel** [comment]
Awesome, thanks a lot @ichiriac! 😃
